lib/ansible/modules/system/lvol_thin.py

```diff
@@ -0,0 +1,755 @@
+        pvs = ''
+
+        lvcreate_cmd = module.get_bin_path("lvcreate", required=True)
+
+        cmd = "%s %s -n %s -V %s%s %s %s %s" % (lvcreate_cmd, yes_opt, lv_name, requested_size, unit, opts, vg_name, pvs)
+
+        rc, _, err = module.run_command(cmd)
+
+        if rc != 0:
+            module.fail_json(msg="Creating logical volume '%s' failed" % lv_name, rc=rc, err=err)
+
+    def convert_to_normal(self, force, opts, pvs):
+        self.delete_lv(self.module, force, self.vg.name, self.name)
+        Lvol.create_lv(self.module, self.size, self.unit, opts, self.vg.name, self.name, pvs)
+
+        return Lvol(self.module, self.vg.name, self.name, self.unit, self.opts, self.pvs)
+
+    def convert_to_thin_pool(self, force):
+        lvol = self.convert_to_normal(force, self.opts, self.pvs)
+        thinPoolLvol = lvol.convert_to_thin_pool(force)
+
+        return thinPoolLvol
+
+    def change_pool(self, pool, force, opts, pvs):
+        lvol = self.convert_to_normal(force, opts, pvs)
+        lvol.convert_to_thin(pool)
+
+        self.__init__(self.module, self.vg.name, self.name, self.unit, self.opts, self.pvs)
+
+
+def validate_size(module, size):
+    if '%' in size:
+        size_parts = size.split('%', 1)
+        size_percent = int(size_parts[0])
+
+        if size_percent > 100:
+            module.fail_json(msg="Size percentage cannot be larger than 100%")
+
+        size_whole = size_parts[1]
+
+        if size_whole == 'ORIGIN':
+            module.fail_json(msg="Snapshot Volumes are not supported")
+        elif size_whole not in ['VG', 'PVS', 'FREE']:
+            module.fail_json(msg="Specify extents as a percentage of VG|PVS|FREE")
+
+    if '%' not in size:
+        if size[-1].lower() in 'bskmgtpe':
+            size = size[0:-1]
+        else:
+            module.fail_json(msg="Invalid units for size: %s" % size)
+
+        try:
+            float(size)
+            if not size[0].isdigit():
+                raise ValueError()
+        except ValueError:
+            module.fail_json(msg="Bad size specification of '%s'" % size)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            vg=dict(required=True),
+            lv=dict(required=True),
+            size=dict(type='str'),
+            opts=dict(type='str'),
+            state=dict(choices=["absent", "present"], default='present'),
+            force=dict(type='bool', default='no'),
+            pvs=dict(type='str'),
+            type=dict(choices=["thin", "thin-pool", "normal"], default="normal"),
+            pool=dict(type='str')
+        ),
+        supports_check_mode=True,
+    )
+
+    vg = module.params['vg']
+    lv = module.params['lv']
+    size = module.params['size']
+    opts = module.params['opts']
+    state = module.params['state']
+    force = module.boolean(module.params['force'])
+    pvs = module.params['pvs']
+    type = module.params['type']
+    pool = module.params['pool']
+
+    if size:
```

Comment:
Not sure if relevant but a size of 0 would not be handled here. Change it ot
`if size is not None`?